public/src/components/Home/SidebarComponent.jsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import ArchivedSubredditsComponent from './ArchivedSubredditsComponent';
+
+const ScrollArea = require('react-scrollbar').default;
+
+export default class SidebarComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            searchFormValue: '',
+            archiveFormValue: '',
+        };
+        this.getSubredditsList = this.getSubredditsList.bind(this);
+        this.onChangeSearchForm = this.onChangeSearchForm.bind(this);
+        this.onChangeArchiveForm = this.onChangeArchiveForm.bind(this);
+        this.getArchiveFormClass = this.getArchiveFormClass.bind(this);
+        this.onKeyPressSearchForm = this.onKeyPressSearchForm.bind(this);
+        this.setSearchValue = this.setSearchValue.bind(this);
+    }
+
+    onKeyPressSearchForm(event) {
+        const { searchFormValue } = this.state;
+        const { storeSubredditToArchive } = this.props;
```

Comment:
Add `prop-types` for type checking?

Reply:
fixed https://github.com/YuriyBC/Reddit-archive/commit/22f4d1c2165a81825081e5eb52592fd5d9e6dc38